pkg/hyper/extractor.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package hyper contains HyperContainer implementation of runtime API.
+// TODO: import this package from hyperhq/hyper directly when it move out of integration.
+
+package hyper
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+type StreamExtractor interface {
```

Comment:
Please un-export those methods and types, also add docs for methods and type definitions in this file.